src/components/fields/TraceSelector.js

```diff
@@ -64,17 +68,30 @@ class TraceSelector extends Component {
     } else {
       this.traceOptions = [{label: 'Scatter', value: 'scatter'}];
     }
+    this.fullValue = plotlyTraceToCustomTrace(props.container);
+  }
 
-    this.fullValue = plotlyTraceToCustomTrace(props.fullContainer);
+  setTraceDefaults(container, fullContainer, updateContainer) {
+    if (
+      (container.type === 'scatter' && !container.mode) ||
+      (!container.type && fullContainer.type === 'scatter')
```

Comment:
doesn't look quite right to me... so if `container.type` is missing, we'll set `mode` even if there already is one?
And if `container.type` is missing, isn't `fullContainer.type` *always* going to be `'scatter'`?

I would have thought just `!container.mode && (container.type === 'scatter' || !container.type)`

Reply:
> !container.mode && (container.type === 'scatter' || !container.type)

you're right, changing this